### PR TITLE
docs(Form): 如何在函数组件中拿到 form 实例

### PR DESCRIPTION
**Pull request by chj-damon**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
there is no a demo that shows developers how they can use `wrappedComponentRef` in functional components. So I made this demo.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | update form document FAQ with using wrappedComponentRef in functional component          |
| 🇨🇳 Chinese | 更新Form文档，在FAQ增加如何在函数组件中使用wrappedComponentRef的例子          |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
Added in FAQ instead.

File: components/form/index.en-US.md [hunk 1]
```diff
@@ -275,3 +275,55 @@ validator(rule, value, callback) => {
   }
 }
 ```
+
+### How you can use `wrappedComponentRef` to get form instance in functional component
```
Comment (zombieJ) [inline components/form/index.en-US.md]: ```Why `wrappedComponentRef` get warning with function component?```

File: components/form/index.en-US.md [hunk 2]
```diff
@@ -275,3 +275,55 @@ validator(rule, value, callback) => {
...
+
+you have to combine `forwardRef` with `useImperativeHandle` so you can get form instance by using `wrappedComponentRef`:
```
Comment (zombieJ) [inline components/form/index.en-US.md]: > You should wrap function component with `forwardRef` to pass ref:

**Comment by afc163** [comment]
做一个 codesandbox 吧，这一大段代码略长。

File: components/form/index.en-US.md (hunk 2 above)
Comment (zombieJ) [reply components/form/index.en-US.md]: `useImperativeHandle` is not must have. Please help to check this.

File: components/form/index.en-US.md [hunk 3]
```diff
@@ -275,3 +275,55 @@ validator(rule, value, callback) => {
...
+import React, { createRef, forwardRef, useImperativeHandle, useState } from 'react';
+import { Form, Input, Button } from 'antd';
+import { FormComponentProps } from 'antd/lib/form/Form';
+
+interface FCFormProps extends FormComponentProps {
+  onSubmit: () => void;
+}
+
+type Ref = FormComponentProps;
+const FCForm = forwardRef<Ref, FCFormProps>(({ form, onSubmit }, ref) => {
```
Comment (zombieJ) [inline components/form/index.en-US.md]: Code block should be simplify. Only show how to use `forwardRef` instead.

File: components/form/index.zh-CN.md [hunk 4]
```diff
@@ -277,3 +277,55 @@ validator(rule, value, callback) => {
   }
 }
 ```
+
+### 如何在函数组件中也能正常使用`wrappedComponentRef`拿到 form 实例
```
Comment (afc163) [inline components/form/index.zh-CN.md]:
```suggestion
### 如何在函数组件中也能正常使用 `wrappedComponentRef` 拿到 form 实例？
```

File: components/form/index.zh-CN.md [hunk 5]
```diff
@@ -277,3 +277,55 @@ validator(rule, value, callback) => {
...
+
+你需要通过`forwardRef`和`useImperativeHandle`的组合使用来实现在函数组件中正确拿到 form 实例：
```
Comment (afc163) [inline components/form/index.zh-CN.md]:
```suggestion
你需要通过 `forwardRef` 和 `useImperativeHandle` 的组合使用来实现在函数组件中正确拿到 form 实例：
```

File: components/form/index.en-US.md (hunk 2 above)
Comment (chj-damon) [reply components/form/index.en-US.md]: https://reactjs.org/docs/hooks-reference.html#useimperativehandle

**Comment by chj-damon** [comment]
> 做一个 codesandbox 吧，这一大段代码略长。
好的。

File: components/form/index.en-US.md (hunk 1 above)
Comment (chj-damon) [reply components/form/index.en-US.md]: warning: `Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()`

File: components/form/index.en-US.md (hunk 3 above)
Comment (chj-damon) [reply components/form/index.en-US.md]: I'll make an codesandbox instead of documenting it.

File: components/form/index.en-US.md (hunk 1 above)
Comment (zombieJ) [reply components/form/index.en-US.md]: I mean change the title to:

> Why `wrappedComponentRef` get warning with function component?

File: components/form/index.en-US.md (hunk 2 above)
Comment (zombieJ) [reply components/form/index.en-US.md]: `useImperativeHandle` is aim to customize `ref` content. `wrappedComponentRef` get the form instance with Form.create. They not use customize `ref` indeed.

Comment (chj-damon) [reply components/form/index.en-US.md]: well, you might be right, but in fact, if I remove `useImperativeHandle`, I cannot get the instance. you can try the codesandbox.

**Comment by zombieJ** [comment]
中英文解释的不是同一件事情，一个是解决 FC warning，一个是拿 form 实例。需要统一一下。

File: components/form/index.en-US.md [hunk 6]
```diff
@@ -275,3 +275,40 @@ validator(rule, value, callback) => {
   }
 }
 ```
+
+### get form instance in functional component
```
Comment (zombieJ) [inline components/form/index.en-US.md]:
```diff
--- get form instance in functional component
+++ Get form instance from functional component
```

File: components/form/index.en-US.md [hunk 7]
```diff
@@ -275,3 +275,40 @@ validator(rule, value, callback) => {
...
+
+you need to combine `forwardRef` with `useImperativeHandle` to get form instance:
```
Comment (zombieJ) [inline components/form/index.en-US.md]: `y` => `Y`

> You can combine ...

File: components/form/index.en-US.md [hunk 8]
```diff
@@ -275,3 +275,40 @@ validator(rule, value, callback) => {
...
+
+your custom form component should look like this:
```
Comment (zombieJ) [inline components/form/index.en-US.md]: `y` => `Y`

File: components/form/index.en-US.md [hunk 9]
```diff
@@ -275,3 +275,40 @@ validator(rule, value, callback) => {
...
+  return (
+    <EnhancedFCForm
+      onSubmit={() => console.log(formRef.current!.form.getFieldValue('name'))}
+      wrappedComponentRef={formRef}
+    />
+  );
+};
+```
+
+you can click the codesandbox to see the complete demo:
```
Comment (zombieJ) [inline components/form/index.en-US.md]: `y` => `Y`

File: components/form/index.en-US.md [hunk 10]
```diff
@@ -275,3 +275,40 @@ validator(rule, value, callback) => {
...
+  return (
+    <EnhancedFCForm
+      onSubmit={() => console.log(formRef.current!.form.getFieldValue('name'))}
+      wrappedComponentRef={formRef}
+    />
+  );
+};
+```
+
+You can click the codesandbox to see the complete demo:
```
Comment (zombieJ) [inline components/form/index.en-US.md]: Online demo:

File: components/form/index.zh-CN.md [hunk 11]
```diff
@@ -277,3 +277,40 @@ validator(rule, value, callback) => {
...
+  return (
+    <EnhancedFCForm
+      onSubmit={() => console.log(formRef.current!.form.getFieldValue('name'))}
+      wrappedComponentRef={formRef}
+    />
+  );
+};
+```
+
+你可以点击下面的 codeSandbox 链接查看完整示例：
```
Comment (zombieJ) [inline components/form/index.zh-CN.md]: 在线示例：

File: components/form/index.en-US.md [hunk 12]
```diff
@@ -275,3 +275,40 @@ validator(rule, value, callback) => {
   }
 }
 ```
+
+### Get form instance in functional component
+
+You need to combine `forwardRef` with `useImperativeHandle` to get form instance:
+
+Your custom form component should look like this:
```
Comment (zombieJ) [inline components/form/index.en-US.md]: 上面有冒号了，这行又一个冒号。感觉可以删掉。中文同。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**